app/javascript/components/ui/Table.tsx

```diff
@@ -0,0 +1,151 @@
+    {children}
+  </thead>
+);
+
+export const TableBody = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableSectionElement>) => {
+  const { busy } = useTable();
+  return (
+    <tbody
+      className={classNames(
+        "contents lg:table-row-group lg:rounded-sm lg:bg-background",
+        busy && "pointer-events-none opacity-50",
+        className,
+      )}
+      {...props}
+    >
+      {children}
+    </tbody>
+  );
+};
+
+export const TableFooter = ({ className, children, ...props }: React.HTMLAttributes<HTMLTableSectionElement>) => {
+  const { busy } = useTable();
+  return (
+    <tfoot
+      className={classNames(
+        "contents font-bold lg:table-footer-group",
+        busy && "pointer-events-none opacity-50",
+        className,
+      )}
+      {...props}
+    >
+      {children}
+    </tfoot>
+  );
+};
+
+export const TableRow = ({
+  className,
+  selected,
+  children,
+  ...props
+}: Omit<React.HTMLAttributes<HTMLTableRowElement>, "aria-selected"> & { selected?: boolean; footer?: boolean }) => (
+  <tr
+    aria-selected={selected}
+    className={classNames(
+      "block rounded-sm border border-border bg-background lg:table-row lg:border-0 lg:bg-transparent",
+      selected && "cursor-pointer hover:bg-active-bg",
+      className,
+    )}
+    {...props}
+  >
+    {children}
+  </tr>
+);
+
+export const TableHead = ({
+  className,
+  sortDirection,
+  onSort,
+  children,
+  ...props
+}: Omit<React.ThHTMLAttributes<HTMLTableCellElement>, "aria-sort"> & {
+  sortDirection?: "ascending" | "descending" | "none";
+  onSort?: () => void;
+}) => (
+  <th
+    aria-sort={sortDirection}
+    onClick={onSort}
+    className={classNames(
+      "px-4 py-3 text-left align-middle lg:table-cell lg:whitespace-nowrap",
+      sortDirection && "cursor-pointer",
+      className,
+    )}
+    {...props}
+  >
+    <span className="inline-flex items-center gap-1">
+      {children}
+      {sortDirection && sortDirection !== "none" ? (
+        <span className="inline-block">{sortDirection === "ascending" ? "↑" : "↓"}</span>
+      ) : null}
+    </span>
+  </th>
+);
+
+export const TableCell = ({
+  className,
+  busy,
+  actions,
+  label,
+  isIcon,
+  children,
+  ...props
+}: Omit<React.TdHTMLAttributes<HTMLTableCellElement>, "aria-busy"> & {
+  busy?: boolean;
+  actions?: boolean;
+  label?: string;
+  isIcon?: boolean;
+}) => (
+  <td
+    aria-busy={busy}
+    className={classNames(
+      "block p-4 text-left align-middle lg:table-cell lg:border-t lg:border-border lg:before:content-none [&:not(:first-child)]:border-t [&:not(:first-child)]:border-border",
+      isIcon && "text-center text-xl lg:w-20 lg:min-w-20 lg:border-r lg:border-border",
+      className,
+    )}
+    {...props}
+  >
+    {label ? <div className="mb-2 font-bold lg:hidden">{label}</div> : null}
+    {actions ? <div className="flex flex-wrap gap-3 lg:justify-end">{children}</div> : children}
```

Comment:
Actions could be a sub-component, but we use that pattern so much a prop seemed more convenient